lib/internal/freelist.js

```diff
@@ -9,18 +10,29 @@ class FreeList {
     this.name = name;
     this.ctor = ctor;
     this.max = max;
-    this.list = [];
+    this.list = ObjectSetPrototypeOf([], null);
```

Comment:
@ExE-Boss I see in https://github.com/nodejs/node/pull/36600 you using SafeArray:
https://github.com/ExE-Boss/node/blob/283189fcce72f98f55d335292db0b3b77a252279/lib/internal/freelist.js#L13

To me it seems the intention is the same to ensures that the array used is not affected by any potential modifications to the standard Array.prototype in user code or other libraries. I think it's great, and this is about maintaining encapsulation and avoiding unexpected side effects caused by changes to global prototypes.

Could you please kindly clarify why we cannot use `SafeArray` here, I'm trying to understand the difference?

Reply:
The difference is largely performance, as <https://github.com/nodejs/node/pull/36600> has a `-91%` perf regression, and this only has a `-90%`.